### PR TITLE
Create one transport certificates Secret per StatefulSet

### PR DESCRIPTION
**Pull request by barkbay**
With this PR the transport certificates are stored in one Secret per `nodeSet`/`StatefulSet`.

* Current/legacy transport certs `Secret` is deleted only when the StatefulSets specifications have been updated (I did some tests and it seems to be fine to delete a Secret which is used by a Pod, the content is still available until the Pod is deleted).
* Unit tests have been added in order to test the new behaviour.
* Transport certs `Secrets` are named `*-transport-certs`  and not `*-transport-certificates` to save a few characters and be consistent with other `Secrets` (`*-[http|transport]-certs-public`,`*-[http|transport]-certs-internal`)

Fixes (or only mitigates ?) #3734

### CODE REVIEW

**Review by pebrc — COMMENTED** [review]
Looks good to me! Much less invasive change than I initially thought. Is it worth backporting this to 1.3?

File: pkg/controller/elasticsearch/driver/downscale.go [hunk 1]
```diff
@@ -161,6 +162,11 @@ func deleteStatefulSetResources(k8sClient k8s.Client, es esv1.Elasticsearch, sta
 		return err
 	}
 
+	err = transport.DeleteStatefulSetTransportCertificate(k8sClient, es.Namespace, statefulSet.Name)
```
[the remark this answers is not in the document]
Comment (barkbay) [reply pkg/controller/elasticsearch/driver/downscale.go:165]: It was my original intention but the `Secrets` are created before the `StatefulSets`. We can find a way to set the ownership later, not sure if it's worth it though ?

[the remark this answers is not in the document]
Comment (pebrc) [reply pkg/controller/elasticsearch/driver/downscale.go:165]: You are right. I did not think of that. Agreed not worth it.

**Review by pebrc — APPROVED** [review]
LGTM! I did some tests and could not find a flaw.

**Comment by barkbay** [comment]
> Is it worth backporting this to 1.3?

I think I would be 👍 once I see all the e2e tests green.
Any thoughts on that @david-kow ?

**Pull request merged**